Paralayout/ViewDistributionItem.swift

```diff
@@ -0,0 +1,334 @@
+//
+//  Copyright © 2021 Square, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import UIKit
+
+/// An element of a horizontal or vertical distribution.
+public enum ViewDistributionItem: ViewDistributionSpecifying {
+
+    /// A UIView, with adjustments to how much space it should take up.
+    case view(UIView, UIEdgeInsets)
+
+    /// A constant spacer between two other elements.
+    case fixed(CGFloat)
+
+    /// Proportional spacer, a fraction of the space not taken up by UIViews or fixed spacers.
+    case flexible(CGFloat)
+
+    // MARK: - Public Static Methods
+
+    /// Filter invisible views (`nil`, uninstalled, hidden, or transparent) from a distribution, and collapse adjacent
+    /// spacers (preferring larger ones).
+    ///
+    /// - parameter distribution: An array of optional distribution specifiers to be collapsed.
+    /// - returns: An array of `DistributionItem`s, without any invisible views or sequential `.fixed` or `.flexible`
+    /// spacers.
+    public static func collapsing(_ distribution: [ViewDistributionSpecifying?]) -> [ViewDistributionItem] {
+        var collapsedItems = [ViewDistributionItem]()
+
+        func isViewVisible(_ view: UIView) -> Bool {
```

Comment:
[nit] This can be a private extension on UIView.

Reply:
The behavior of this entire method is a bit confusing, so I think we're going to change it up. Still an open discussion around exactly how it should work though. See #29.